gapis/server/server.go

```diff
@@ -599,11 +600,22 @@ func (l *statusListener) OnMemorySnapshot(ctx context.Context, stats runtime.Mem
 	})
 }
 
-func (s *server) Status(ctx context.Context, snapshotInterval, statusInterval time.Duration, f func(*service.TaskUpdate), m func(*service.MemoryStatus)) error {
+func (l *statusListener) OnReplayStatusUpdate(ctx context.Context, label uint64, total_instrs uint32, finished_instrs uint32) {
+	l.progressMutex.Lock()
```

Comment:
Can I delete these two lines to send replay status update without mutex lock on? Want to avoid lock for efficiency, will do the progress non-decreasing check at client side.

Reply:
I would say yes - it looks like the mutex is only there to guard `lastProgressUpdate` and `progressUpdateFreq`.
There's a whole bunch of locking and unlocking in other functions that looks redundant and could be removed.
I'd suggest commenting the fields that they belong to the mutex, so this is less confusing in the future:

```go
	lastProgressUpdate map[*status.Task]time.Time // guarded by progressMutex
	progressUpdateFreq time.Duration // guarded by progressMutex
	progressMutex      sync.Mutex
```

Reply:
Thank you Ben for answering! It's of great help! Will add the comments.